### PR TITLE
remove movedim.intlist, tensor_split*, to.* from xfail list

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #110999

### CODE REVIEW

File: aten/src/ATen/functorch/LegacyBatchingRegistrations.cpp [hunk 1]
```diff
@@ -747,8 +735,6 @@ TORCH_LIBRARY_IMPL(_, FuncTorchBatched, m) {
 
 TORCH_LIBRARY_IMPL(aten, FuncTorchBatched, m) {
   // still legacy b/c teturns multiple tensors
-  m.impl("tensor_split.sections", tensor_split_sections_batching_rule);
-  m.impl("tensor_split.indices", tensor_split_indices_batching_rule);
```
Comment (kshitij12345) [inline aten/src/ATen/functorch/LegacyBatchingRegistrations.cpp:751]: We can probably remove `tensor_split_indices_batching_rule` as well.

File: aten/src/ATen/functorch/BatchRulesBinaryOps.cpp [hunk 2]
```diff
@@ -200,11 +200,15 @@ static std::tuple<Tensor,optional<int64_t>> masked_select_batch_rule(
   return std::make_tuple(result, 0);
 }
 
-static std::tuple<Tensor,optional<int64_t>> masked_select_backward_batch_rule(
-    const Tensor& grad, optional<int64_t> grad_bdim,
-    const Tensor& self, optional<int64_t> self_bdim,
-    const Tensor& mask, optional<int64_t> mask_bdim) {
-  TORCH_CHECK(!mask_bdim.has_value(),
+static std::tuple<Tensor, optional<int64_t>> masked_select_backward_batch_rule(
```
Comment (kshitij12345) [inline aten/src/ATen/functorch/BatchRulesBinaryOps.cpp:203]: This is a just a formatting change. I think the change to remove this function and add a corresponding decomposition was missed in the commit.

**Review by kshitij12345 — COMMENTED** [review]
Minor comments, will wait till the `masked_select_backward` comment is addressed to approve.

Thank you @guilhermeleobas

File: aten/src/ATen/functorch/BatchRulesBinaryOps.cpp (hunk 2 above)
Comment (guilhermeleobas) [reply aten/src/ATen/functorch/BatchRulesBinaryOps.cpp:203]: Does `lintrunner` applies to C/C++ files as well?

Comment (kshitij12345) [reply aten/src/ATen/functorch/BatchRulesBinaryOps.cpp:203]: Yes, IIRC correctly it runs clang-tidy on files under `aten`.

Comment (guilhermeleobas) [reply aten/src/ATen/functorch/BatchRulesBinaryOps.cpp:203]: Thanks. I guess this formatting change was done by my vscode setup. Will undo this change

Comment (kshitij12345) [reply aten/src/ATen/functorch/BatchRulesBinaryOps.cpp]: We should update the PR title to remove `masked_select_backward` and `onehot` as those are not handled in this PR.

Comment (guilhermeleobas) [reply aten/src/ATen/functorch/BatchRulesBinaryOps.cpp]: Done

**Review by kshitij12345 — APPROVED** [review]
LGTM! Thanks @guilhermeleobas

**Comment by guilhermeleobas** [comment]
@kshitij12345, can I merge this one or does it needs a second review?

**Comment by kshitij12345** [comment]
It should be ok to merge this.

**Pull request closed without merging**